### DISCUSSION TITLE
How to eable multicast traffic in site-to-site VPN

### DISCUSSION THREAD

**Discussion by lkpatel** [strongswan/strongswan#2453] [Q&A]
I have a use case where a user at one site is doing multicast (using VLC player in test setup) and users at other site have to receive it. I recompiled the strongswan for enabling forecast. I tried example given in documentation for Roadwarrior case, but same configuration setup is not working in site to site VPN case. I am running a VLC media player at 10.10.10.11 to multicast at address 239.255.0.1 and trying to listen it on 10.10.11.11 . I have enabled forwarding in both gateway i.e. Sun and Moon.

![Screenshot 2024-09-17 175707](https://github.com/user-attachments/assets/ea955649-6d18-478a-a128-25b566e43717)
Config of swanctl.conf @ Sun:

> connections {
>     net {
>       local_addrs  = 4.4.4.1
>       remote_addrs = 4.4.4.2
>       local {
>         auth = pubkey
>         certs = sunCert.pem
>       }
>       remote {
>         auth = pubkey
>         id = "C=CH, O=strongSwan, CN=moon.strongswan.org"
>       }
>       children {
>         net-net {
>           local_ts  = 10.10.10.0/24,224.0.0.0/4
>           remote_ts = dynamic,10.10.11.0/24,224.0.0.0/4
>           start_action = trap|start
>           esp_proposals = aes128-sha256-modp3072
>         }
>       }
>       version = 2
>       proposals = aes128-sha256-modp3072
>       dpd_delay = 14
>     }
>   }

Config at strongswan.conf @ Sun:

> charon {
>         load = random nonce aes sha1 sha2 pem pkcs1 curve25519 gmp x509 curl revocation hmac kdf vici kernel-netlink socket-default updown attr forecast
>         load_modular = yes
>         plugins {
>                 include strongswan.d/charon/*.conf
>                 forecast {
>                         groups = 224.0.0.1,224.0.0.22,224.0.0.251,224.0.0.252,239.255.255.250,239.255.0.1
>                         interface = eno3
>                         reinject = net
>                  }
>         }
> }

Config of swanctl.conf @ Moon:

> connections {
>     net {
>       local_addrs  = 4.4.4.2
>       remote_addrs = 4.4.4.1
>       local {
>         auth = pubkey
>         certs = moonCert.pem
>       }
>       remote {
>         auth = pubkey
>         id = "C=CH, O=strongSwan, CN=sun.strongswan.org"
>       }
>       children {
>         net-net {
>           local_ts  = 10.10.11.0/24,224.0.0.0/4
>           remote_ts = 10.10.10.0/24,224.0.0.0/4
>           start_action = trap|start
>           esp_proposals = aes128-sha256-modp3072
>         }
>       }
>       version = 2
>       proposals = aes128-sha256-modp3072
>       dpd_delay = 12
>     }
>   }

Config of strongswan.conf @ moon

> charon {
>         load = random nonce aes sha1 sha2 pem pkcs1 curve25519 gmp x509 curl revocation hmac kdf vici kernel-netlink socket-default updown attr forecast
>         load_modular = yes
>         plugins {
>                 include strongswan.d/charon/*.conf
>                 forecast {
>                         groups = 224.0.0.1,224.0.0.22,224.0.0.251,224.0.0.252,239.255.255.250,239.255.0.1
>                         interface = eno3
>                         #reinject = net
>                  }
>         }
> }

Could anybody point out at where I am wrong ?

**Comment by tobiasbrunner** [comment]
Please read the [documentation for the plugin](https://docs.strongswan.org/docs/5.9/plugins/forecast.html) (you are definitely missing the mark).

**Reply by lkpatel** [reply]
When I enable 

> mark_in  = %unique
> mark_out = %unique

  in swanctl.conf, I am seeing error "**_initializing iptables failed: iptables who? (do you need to insmod?)_**"

My OS: Rocky Linux release 9.3 (Blue Onyx)
uname -a : Linux sun.strongswan 5.14.0-362.24.1.el9_3.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Mar 13 17:33:16 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

I had installed following rpm-

> iptables-libs-1.8.8-6.el9_1.x86_64
> iptables-nft-1.8.8-6.el9_1.x86_64
> iptables-legacy-libs-1.8.10-2.2.el9.x86_64
> iptables-legacy-devel-1.8.10-2.2.el9.x86_64

**Reply by tobiasbrunner** [reply]
Note that these packages only install the libs, possibly not the module. But could be a problem with your kernel. Or you literally have to actually load the module manually. If only nftables is available, the plugin won't work, it currently only supports iptables (and even if both are loaded it might not work correctly if the rest of the firewall rules are configured via nftables).
